tests/credentials_test.go

```diff
@@ -292,33 +292,40 @@ var _ = Describe("[sig-compute]Guest Access Credentials", decorators.SigCompute,
 			Eventually(matcher.ThisVMI(vmi), 240*time.Second, 2*time.Second).Should(matcher.HaveConditionTrue(v1.VirtualMachineInstanceUnsupportedAgent))
 		})
 	})
-	Context("with secret and configDrive propagation", func() {
-		It("[test_id:6224]should have ssh-key under authorized keys", func() {
-			secretID := "my-pub-key"
-			userData := fmt.Sprintf(
-				"#cloud-config\npassword: %s\nchpasswd: { expire: False }\n",
-				fedoraPassword,
-			)
-			vmi := tests.NewRandomVMIWithEphemeralDiskAndConfigDriveUserdataHighMemory(cd.ContainerDiskFor(cd.ContainerDiskFedoraTestTooling), userData)
-			vmi.Namespace = util.NamespaceTestDefault
-			vmi.Spec.AccessCredentials = []v1.AccessCredential{
-				{
-					SSHPublicKey: &v1.SSHPublicKeyAccessCredential{
-						Source: v1.SSHPublicKeyAccessCredentialSource{
-							Secret: &v1.AccessCredentialSecretSource{
-								SecretName: secretID,
-							},
-						},
-						PropagationMethod: v1.SSHPublicKeyAccessCredentialPropagationMethod{
-							ConfigDrive: &v1.ConfigDriveSSHPublicKeyAccessCredentialPropagation{},
-						},
-					},
-				},
-			}
+	Context("with secret and cloudInit propagation", func() {
+		var vmi *v1.VirtualMachineInstance
+		secretID := "my-pub-key"
+		userData := fmt.Sprintf(
+			"#cloud-config\npassword: %s\nchpasswd: { expire: False }\n",
+			fedoraPassword,
+		)
+
+		key1 := "ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkT test-ssh-key1"
+		key2 := "ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkT test-ssh-key2"
+		key3 := "ssh-rsa AAAAB3NzaC1yc2EAAAABIwAAAQEA6NF8iallvQVp22WDkT test-ssh-key3"
+
+		verifySSHKeys := func(vmi *v1.VirtualMachineInstance) {
```

Comment:
What's stopping us from using an actual keypair here and ssh'ing into the VMI?

Reply:
Nothing I guess, but let's do optimizations in separate PR. I just extended previous test for configDrive. 
I have no proper testing environment and time for developing this, sorry